Error when querying app_kpop_group during seed

```
2024-03-09T16:15:36.428Z [ERROR] - interactionCreate | gid: 1214719939446050936, tid: 1214719939936522243 | Error while invoking command (CHAT_INPUT CommandInteraction interaction for 'exclude') | 402b6157-8a93-4bcb-92b7-ceb7871e0dcb |  Data: {"id":"1213407215453732930","name":"exclude","options":[{"name":"add","options":[{"name":"group_1","type":3,"value":"ATEEZ"}],"type":1}],"type":1} | Exception Name: Error. Reason: Unknown column 'has_songs' in 'where clause'. Trace: Error: Unknown column 'has_songs' in 'where clause'
    at Packet.asError (/app/node_modules/mysql2/lib/packets/packet.js:728:17)
    at Query.execute (/app/node_modules/mysql2/lib/commands/command.js:29:26)
    at PoolConnection.handlePacket (/app/node_modules/mysql2/lib/connection.js:481:34)
    ```
    
 Race condition between when we import daisuki seed and when we generate our data tables, that column is missing

--- a/sql/procedures/create_kmq_data_tables_procedure.sql
+++ b/sql/procedures/create_kmq_data_tables_procedure.sql
@@ -125,11 +125,21 @@ BEGIN
 	RENAME TABLE available_songs TO old, available_songs_temp TO available_songs;
 	DROP TABLE old;
 
-	/* mark artists as not having songs */
-	ALTER TABLE kpop_videos.app_kpop_group ADD COLUMN IF NOT EXISTS has_songs TINYINT(1) DEFAULT 0;
-	UPDATE kpop_videos.app_kpop_group
+	/* atomically swap app_kpop_group to avoid race condition*/
+	DROP TABLE IF EXISTS kpop_videos.app_kpop_group_temp;
+	CREATE TABLE IF NOT EXISTS kpop_videos.app_kpop_group_temp LIKE kpop_videos.app_kpop_group;
+	INSERT INTO kpop_videos.app_kpop_group_temp
+	SELECT * FROM kpop_videos.app_kpop_group;
+
+	/* mark artists as not having songs */	
+	ALTER TABLE kpop_videos.app_kpop_group_temp ADD COLUMN IF NOT EXISTS has_songs TINYINT(1) DEFAULT 0;
+	UPDATE kpop_videos.app_kpop_group_temp
 	SET has_songs = 1
 	WHERE id in (SELECT DISTINCT(id_artist) FROM available_songs
-	RIGHT JOIN kpop_videos.app_kpop_group ON available_songs.id_artist = kpop_videos.app_kpop_group.id);
+	RIGHT JOIN kpop_videos.app_kpop_group_temp ON available_songs.id_artist = kpop_videos.app_kpop_group_temp.id);
+
+	RENAME TABLE kpop_videos.app_kpop_group TO old, kpop_videos.app_kpop_group_temp TO kpop_videos.app_kpop_group;
+	DROP TABLE old;
+
 END //
 DELIMITER ;

